pkg/asset/installconfig/azure/client.go

```diff
@@ -152,3 +155,28 @@ func (c *Client) getProvidersClient(ctx context.Context) (azres.ProvidersClient,
 	client.Authorizer = c.ssn.Authorizer
 	return client, nil
 }
+
+// ValidateResourceSkuAvailability checks if the diskType exists for the given region.
+func (c *Client) ValidateResourceSkuAvailability(ctx context.Context, region string, diskType string) (*azsku.ResourceSku, error) {
+	client := azsku.NewResourceSkusClient(c.ssn.Credentials.SubscriptionID)
+	client.Authorizer = c.ssn.Authorizer
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	resourceSkus, err := client.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, page := range resourceSkus.Values() {
+		if to.String(page.Name) == diskType {
+			for _, diskRegion := range to.StringSlice(page.Locations) {
+				if diskRegion == region {
+					return &page, nil
+				}
+			}
+		}
+	}
+
+	return nil, errors.New("ultraSSD_LRS not available for specified subscription")
```

Comment:
It would be better to use `diskType` var in error message. Perhaps `region` as well